### PR TITLE
fix: show correct url in output status

### PR DESCRIPTION
**Pull request by snitin315**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
Yes

### Motivation / Use-Case

https://github.com/webpack/webpack-dev-server/pull/2992#discussion_r571360196

Screenshot at 2021-02-14 12-43-58

### Breaking Changes

None.

### Additional Info
No

### CODE REVIEW

File: lib/utils/normalizeOptions.js [hunk 1]
```diff
@@ -73,6 +73,9 @@ function normalizeOptions(compiler, options) {
   options.liveReload =
     typeof options.liveReload !== 'undefined' ? options.liveReload : true;
 
+  options.host =
+    typeof options.host !== 'undefined' ? options.host : 'localhost';
```
Comment (rishabh3112) [inline lib/utils/normalizeOptions.js:77]:
```suggestion
    typeof options.host !== 'undefined' ? options.host : null;
```

**Review by rishabh3112 — CHANGES_REQUESTED** [review]
Needs update in snapshots as well.

File: lib/utils/normalizeOptions.js (hunk 1 above)
Comment (snitin315) [reply lib/utils/normalizeOptions.js]: `null` won't fix the issue here - https://github.com/webpack/webpack-dev-server/pull/2992#discussion_r571360196 . I'm still unsure about this approach

Comment (rishabh3112) [reply lib/utils/normalizeOptions.js]: Thanks for updating issue.
My suggestion was based on test here https://github.com/webpack/webpack-dev-server/blob/master/test/server/host-option.test.js

Ideally, https://github.com/webpack/webpack-dev-server/blob/master/test/server/host-option.test.js#L18 and https://github.com/webpack/webpack-dev-server/blob/master/test/server/host-option.test.js#L48 should have same behaviour. But that maybe a breaking change.

**Review by alexander-akait — DISMISSED** [review]

**Comment by alexander-akait** [comment]
Please fix lint problem

**Review by alexander-akait — APPROVED** [review]

**Comment by snitin315** [comment]
I shouldn't be able to commit if there is a lint issue until I bypass with `--no-verify` 😕 .

**Comment by snitin315** [comment]
I will fix ci tomorrow.

**Review by alexander-akait — COMMENTED** [review]
Can you test locally? Using a lot of CI reduce available machines for other projects

**Comment by snitin315** [comment]
Sorry about that, sometimes it passes locally but fails on CI. I will keep this in my mind and will use my fork's CI in the future in such a case. 

This one should be good now.

File: lib/Server.js [hunk 2]
```diff
@@ -588,14 +588,18 @@ class Server {
...
     const prettyPrintUrl = (newHostname) =>
       url.format({ protocol, hostname: newHostname, port, pathname: '/' });
 
     let prettyHostname;
     let lanUrlForTerminal;
 
-    if (hostname === '0.0.0.0' || hostname === '::') {
+    if (
+      hostname === '0.0.0.0' ||
+      hostname === '127.0.0.1' ||
```
Comment (alexander-akait) [inline lib/Server.js:600]: In theory `127.0.0.1` can be not localhost (very rare edge case), also localhost can be not `127.0.0.1` (also very rare edge case). Why we added this?

Comment (snitin315) [reply lib/Server.js:600]: because without it we use `webpack serve --host localhost` it was printing `server running at http://127.0.0.1:8080` instead of `http://localhost:8080`. also failing some snapshots. I thought they should be the same.

Screenshot at 2021-02-15 19-03-26

Comment (alexander-akait) [reply lib/Server.js:600]: Here right solution https://github.com/indutny/node-ip/blob/master/lib/ip.js#L320 just use `ip.isLoopback()`

Comment (alexander-akait) [reply lib/Server.js:600]: Should be easy to fix :star:

Comment (snitin315) [reply lib/Server.js:600]: Like this ?

```suggestion
      ip.isLoopback(hostname) ||
```

Comment (alexander-akait) [reply lib/Server.js:600]: Yep, we also can remove `hostname === '::'` and maybe `0.0.0.0` need test it (manually as minimum)

**Review by alexander-akait — APPROVED** [review]

**Comment by alexander-akait** [comment]
Thanks

File: test/cli/cli.test.js [hunk 3]
```diff
@@ -6,42 +6,7 @@ const internalIp = require('internal-ip');
...
 
-// skip if webpack-dev-server is not linked
-let runCLITest = describe;
-let basePath;
-
-try {
-  basePath = path.join(require.resolve('webpack-dev-server'), '..', '..');
-} catch {
-  runCLITest = describe.skip;
-}
```
Comment (ylemkimon) [inline test/cli/cli.test.js:17]: This'd make tests fail where `webpack-dev-server` is not `npm link`ed.

Comment (snitin315) [reply test/cli/cli.test.js:17]: We link it on CI before running tests.

Comment (ylemkimon) [reply test/cli/cli.test.js:17]: But for developers testing locally, it may be a hurdle.
